backend/src/test/java/de/neuefische/booklibrary/backend/BookServiceTest.java

```diff
@@ -31,20 +30,51 @@ void getAllBooks_ifNoBooksExist_returnEmptyList() {
     }
 
 
-   @Test
-   void getAllBooks_ifBooksExist_returnListOfAllBooks(){
+    @Test
+    void getAllBooks_ifBooksExist_returnListOfAllBooks() {
         //given
-       List<Book> bookList = new ArrayList<>(List.of(
-               new Book("id1", "Java", "M. Kofler", "978-3-8362-8392-2", BookState.AVAILABLE)
-       ));
-       when(bookRepository.findAll()).thenReturn(bookList);
+        List<Book> bookList = new ArrayList<>(List.of(
+                new Book("id1", "Java", "M. Kofler", "978-3-8362-8392-2", BookState.AVAILABLE)
+        ));
+        when(bookRepository.findAll()).thenReturn(bookList);
+
+        //when
+        List<Book> actual = bookService.getAllBooks();
+
+        //then
+        List<Book> expected = bookList;
+        assertEquals(expected, actual);
 
-       //when
-       List<Book> actual = bookService.getAllBooks();
+    }
 
-       //then
-       List<Book> expected = bookList;
-       assertEquals(expected, actual);
 
-   }
+    @Test
+    void addNewBookWithoutId_returnBookWithId() {
+        //given
+        Book newBook = new Book(null, "Java-Script", "P. Ackermann", "978-3-8362-8629-9", BookState.NOT_AVAILABLE);
+        Book newBookWithId = new Book("id1", "Java-Script", "P. Ackermann", "978-3-8362-8629-9", BookState.NOT_AVAILABLE);
+
+        when(bookRepository.save(newBook)).thenReturn(newBookWithId);
+
+        //when
+        Book actual = bookService.saveBook(newBook);
+
+        //then
+        Book expected = newBookWithId;
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void addNewBookWithId_returnUpdatedBook() {
```

Comment:
I think it's not REST-conform to update an existing element with POST-Endpoint, but if you wan't it to work that way, my above comment about .insert() is obsolete